train data와 test data의 matrix를 비교하는 코드입니다.

In [1]:
import os
import pandas as pd
import numpy as np
os.getcwd()

'/home/hb/python/git_pp_pred/EDA'

In [3]:
pp_prediction/data/after_HMM/dfpsphmm_ratiorandomver2.pickle

NameError: name 'pp_prediction' is not defined

In [2]:
import pickle

In [3]:
with open("/home/hb/python/git_pp_pred/data/(kw)dfpsphmm_ratiorandomver2.pickle", "rb" ) as file:
    psph = pickle.load(file)

In [4]:
psph

array([['Dyrk2', 'DYRK2', 'Q5U4C9', ..., 1,
        list(['L', 'G', 'S', 'S', 'R', 'P', 'S', 'S', 'A', 'P', 'G', 'M', 'L', 'P', 'L']),
        array([[[1.152 , 0.981 , 1.023 , ..., 1.152 , 0.886 , 1.152 ],
                [0.984 , 0.7866, 0.822 , ..., 0.984 , 0.652 , 0.984 ],
                [1.238 , 1.037 , 0.979 , ..., 1.238 , 0.8213, 1.238 ],
                ...,
                [0.    , 0.    , 0.    , ..., 0.    , 0.    , 0.    ],
                [0.    , 0.    , 0.    , ..., 0.    , 0.    , 0.    ],
                [0.    , 0.    , 0.    , ..., 0.    , 0.    , 0.    ]]],
              dtype=float16)                                            ],
       ['Pak2', 'PAK2', 'Q64303', ..., 1,
        list(['R', 'T', 'P', 'G', 'R', 'P', 'L', 'S', 'S', 'Y', 'G', 'M', 'D', 'S', 'R']),
        array([[[1.062 , 1.027 , 0.886 , ..., 0.8477, 1.023 , 1.062 ],
                [0.8774, 1.118 , 0.797 , ..., 0.9165, 1.018 , 0.8774],
                [0.98  , 0.8774, 0.7417, ..., 1.039 , 0.881 , 0.98

In [9]:
import pickle

with open("/home/hb/python/pp_prediction/EDA/EffnetTotal.pickle", "rb" ) as file:
    prediction_value = pickle.load(file)
    
import pandas as pd
import numpy as np

probs = np.stack(prediction_value[0])
probs = pd.DataFrame(probs)
probs.columns = ['probs']

preds = np.stack(prediction_value[1])
preds = pd.DataFrame(preds)
preds.columns = ['preds']

In [10]:
preds

preds
0        0
1        1
2        1
3        1
4        1
..     ...
885      0
886      1
887      0
888      0
889      1

[890 rows x 1 columns]

In [11]:
display(testdata.head(1))

NameError: name 'testdata' is not defined

In [31]:
  # train data CSV파일 읽기
traindata = pd.read_csv("/home/hb/python/pp_prediction/traindata.csv")
traindata.columns

traindata.drop(['Unnamed: 0'], axis=1, inplace=True)
display(traindata.head(1))

  # test data CSV파일 읽기
# # save data
# with open('depod(testdata)_0418.pickle','wb') as fw:
#     pickle.dump(depod, fw)

# load data
with open('/home/hb/python/pp_prediction/test_Data_curation/depod(testdata)_0418.pickle', 'rb') as fr:
    testdata = pickle.load(fr)

display(testdata.head(1))



Substrate_UniprotID Substrate_name    Substrate_seq Kinase  \
0              Q9ERR1          NDEL1  LGSSRPSSAPGMLPL  DYRK2   

                                          Kinase_seq  Length  answer  \
0  YEVLKVIGKGSFGQVVKAYDHKVHQHVALKMVRNEKRFHRQAAEEI...     313       1   

                                              matrix  
0  [[[1.152  0.981  1.023  ... 1.152  0.886  1.15...

AttributeError: Can't get attribute '_unpickle_block' on <module 'pandas._libs.internals' from '/home/hb/anaconda3/lib/python3.9/site-packages/pandas/_libs/internals.cpython-39-x86_64-linux-gnu.so'>

In [47]:
testdata['matrix']

0      [[[0.965  1.207  1.078  ... 0.9233 0.9233 1.20...
1      [[[0.993  0.965  1.078  ... 0.886  1.078  1.03...
2      [[[0.7417 1.137  1.097  ... 1.17   1.17   1.09...
3      [[[1.23   0.886  0.965  ... 1.165  1.229  1.03...
4      [[[0.873  1.042  1.034  ... 1.123  1.042  1.19...
                             ...                        
885    [[[0.87   1.211  0.9473 ... 0.976  1.211  0.88...
886    [[[1.1875 1.269  0.981  ... 0.981  1.269  1.06...
887    [[[1.165  1.023  1.165  ... 1.165  1.023  1.21...
888    [[[1.037  1.165  1.037  ... 0.9087 1.165  0.95...
889    [[[0.88   1.079  1.115  ... 0.861  1.079  1.09...
Name: matrix, Length: 890, dtype: object

In [32]:
testdata[['matrix']]

matrix
0    [[[0.965  1.207  1.078  ... 0.9233 0.9233 1.20...
1    [[[0.993  0.965  1.078  ... 0.886  1.078  1.03...
2    [[[0.7417 1.137  1.097  ... 1.17   1.17   1.09...
3    [[[1.23   0.886  0.965  ... 1.165  1.229  1.03...
4    [[[0.873  1.042  1.034  ... 1.123  1.042  1.19...
..                                                 ...
885  [[[0.87   1.211  0.9473 ... 0.976  1.211  0.88...
886  [[[1.1875 1.269  0.981  ... 0.981  1.269  1.06...
887  [[[1.165  1.023  1.165  ... 1.165  1.023  1.21...
888  [[[1.037  1.165  1.037  ... 0.9087 1.165  0.95...
889  [[[0.88   1.079  1.115  ... 0.861  1.079  1.09...

[890 rows x 1 columns]

In [43]:
# testdata_ = pd.concat([testdata,probs,preds], axis=1)
# testdata_.head(1)

# testdata = testdata_
# testdata.head(3)

# testdata.to_csv("test.csv")


In [45]:
os.getcwd()

'/home/hb/python/pp_prediction/EDA'

In [44]:
# testdata.to_csv("test.csv")


In [33]:
traindata_true = traindata[traindata['answer'] == 1]
traindata_false = traindata[traindata['answer'] == 0]

testdata_true = testdata[testdata['answer'] == 1]
testdata_false = testdata[testdata['answer'] == 0]

In [34]:
list_traintrue_matrix = list(traindata_true.iloc[:,-1])
list_trainfalse_matrix = list(traindata_false.iloc[:,-1])

In [36]:
testdata_true.loc[:,['train']] = 2

for i in range(len(testdata_true)):
    if testdata_true.iloc[i,-2] in list_traintrue_matrix:
        testdata_true.iloc[i,-1] = 1
    else:
        testdata_true.iloc[i,-1] = 0
        
testdata_true.head(10)
        


/home/hb/anaconda3/lib/python3.9/site-packages/pandas/core/indexing.py:1817: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(loc, value, pi)


Substrate_UniprotID    Substrate_name    Substrate_seq Kinase  \
0              O00401      N-WASP, WASL  EITTNRFYGPQVNNI   ABL1   
1              O00401      N-WASP, WASL  RETSKVIYDFIEKTG   ABL1   
2              O14745  NHERF1, SLC9A3R1  PALVRSASSDTSEEL   GRK6   
3              O14920       IKBKB, IKKB  AKELDQGSLCTSFVG   CHUK   
4              O14920       IKBKB, IKKB  DQGSLCTSFVGTLQY   CHUK   
5              O14920       IKBKB, IKKB  ELDQGSLCTSFVGTL  IKBKB   
6              O14920       IKBKB, IKKB  SKVRGPVSGSPDSMN  IKBKB   
7              O14920       IKBKB, IKKB  VRGPVSGSPDSMNAS  IKBKB   
8              O14920       IKBKB, IKKB  PVSGSPDSMNASRLS  IKBKB   
9              O14920       IKBKB, IKKB  SPDSMNASRLSQPGQ  IKBKB   

                                          Kinase_seq  Length  answer  \
0  MKHKLGGGQYGEVYEGVWKKYSLTVAVKTLKEDTMEVEEFLKEAAV...     247       1   
1  MKHKLGGGQYGEVYEGVWKKYSLTVAVKTLKEDTMEVEEFLKEAAV...     247       1   
2  RQYRVLGKGGFGEVCACQVRATGKMYACKKLEKKRIKKRKGEAMAL...     261       1   
3  MRERLGTGGFGNVCLYQHRELDLKIAIKSCRLELSTKNRERWCHEI...     272       1   
4  MRERLGTGGFGNVCLYQHRELDLKIAIKSCRLELSTKNRERWCHEI...     272       1   
5  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       1   
6  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       1   
7  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       1   
8  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       1   
9  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       1   

                                              matrix  train  
0  [[[0.965  1.207  1.078  ... 0.9233 0.9233 1.20...      1  
1  [[[0.993  0.965  1.078  ... 0.886  1.078  1.03...      1  
2  [[[0.7417 1.137  1.097  ... 1.17   1.17   1.09...      1  
3  [[[1.23   0.886  0.965  ... 1.165  1.229  1.03...      0  
4  [[[0.873  1.042  1.034  ... 1.123  1.042  1.19...      0  
5  [[[0.8345 0.984  0.673  ... 0.7866 0.8745 0.98...      0  
6  [[[0.822  1.199  0.9087 ... 0.822  0.965  0.78...      0  
7  [[[0.9087 1.17   0.7866 ... 0.7827 1.012  0.82...      0  
8  [[[0.652  0.9087 0.822  ... 1.17   0.984  0.82...      0  
9  [[[0.822  0.652  0.673  ... 0.652  0.7866 0.89...      0

In [37]:
testdata_false.loc[:,['train']] = 2

for i in range(len(testdata_false)):
    if testdata_false.iloc[i,-2] in list_trainfalse_matrix:
        testdata_false.iloc[i,-1] = 1
    else:
        testdata_false.iloc[i,-1] = 0
        
testdata_false.head(10)
        

/home/hb/anaconda3/lib/python3.9/site-packages/pandas/core/indexing.py:1817: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(loc, value, pi)


Substrate_UniprotID    Substrate_name    Substrate_seq Kinase  \
445              O00401      N-WASP, WASL  EYGYGYPYYYIYGYQ   ABL1   
446              O00401      N-WASP, WASL  PSDSDSTSYSPSKSG   ABL1   
447              O14745  NHERF1, SLC9A3R1  FYKYVYEYPYPYRYG   GRK6   
448              O14920       IKBKB, IKKB  GYPYSYSYSYIYDYL   CHUK   
449              O14920       IKBKB, IKKB  RSPSPSGSPSKSRSA   CHUK   
450              O14920       IKBKB, IKKB  SYLYTYPYNYRYQYP  IKBKB   
451              O14920       IKBKB, IKKB  SSVSLSGSLSASDSP  IKBKB   
452              O14920       IKBKB, IKKB  EYAYDYQYTYLYPYI  IKBKB   
453              O14920       IKBKB, IKKB  RTQTRTQTGTPTITP  IKBKB   
454              O14920       IKBKB, IKKB  TYTYSYNYVYTYRYN  IKBKB   

                                            Kinase_seq  Length  answer  \
445  MKHKLGGGQYGEVYEGVWKKYSLTVAVKTLKEDTMEVEEFLKEAAV...     247       0   
446  MKHKLGGGQYGEVYEGVWKKYSLTVAVKTLKEDTMEVEEFLKEAAV...     247       0   
447  RQYRVLGKGGFGEVCACQVRATGKMYACKKLEKKRIKKRKGEAMAL...     261       0   
448  MRERLGTGGFGNVCLYQHRELDLKIAIKSCRLELSTKNRERWCHEI...     272       0   
449  MRERLGTGGFGNVCLYQHRELDLKIAIKSCRLELSTKNRERWCHEI...     272       0   
450  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       0   
451  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       0   
452  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       0   
453  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       0   
454  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       0   

                                                matrix  train  
445  [[[0.965 1.193 1.034 ... 1.034 1.193 1.042]\n ...      0  
446  [[[0.836  1.     0.873  ... 0.886  1.     1.03...      0  
447  [[[0.942  1.062  0.7046 ... 0.98   1.062  0.91...      0  
448  [[[1.034  1.193  0.836  ... 0.873  1.193  1.12...      0  
449  [[[0.993  1.     0.836  ... 0.993  1.     1.23...      0  
450  [[[0.822  0.865  0.984  ... 0.8994 0.865  0.65...      0  
451  [[[0.822  0.822  0.9087 ... 0.673  0.822  0.65...      0  
452  [[[0.8345 0.865  1.012  ... 0.652  0.865  0.95...      0  
453  [[[1.17   0.8745 0.8994 ... 0.952  0.8745 0.65...      0  
454  [[[0.8745 0.865  0.8745 ... 1.17   0.865  0.78...      0

test data 중 true 값에서 train true data와 얼마나 겹치는지 보기



In [42]:
from sklearn.metrics import roc_auc_score
auc1 = roc_auc_score(testdata[["answer"]], testdata[["probs"]])

auc1

auc2 = roc_auc_score(testdata_non[["answer"]], testdata_non[["probs"]])
auc2

In [38]:
# 트레인과 중복되는 테스트 데이터
testdata_true_anwswer1 = testdata_true[testdata_true['train'] == 1]
display(testdata_true_anwswer1.head(5))
print(len(testdata_true_anwswer1))
print(len(testdata_true))

# 트레인과 중복되지 않는 테스트 데이터
testdata_true_anwswer0 = testdata_true[testdata_true['train'] == 0]
display(testdata_true_anwswer0.head(5))
print(len(testdata_true_anwswer0))
print(len(testdata_true))

Substrate_UniprotID        Substrate_name    Substrate_seq Kinase  \
0               O00401          N-WASP, WASL  EITTNRFYGPQVNNI   ABL1   
1               O00401          N-WASP, WASL  RETSKVIYDFIEKTG   ABL1   
2               O14745      NHERF1, SLC9A3R1  PALVRSASSDTSEEL   GRK6   
28              O60674                  JAK2  VLPQDKEYYKVKEPG   ABL1   
30              O60716  CTNND1, p120 catenin  MEGYRAPSRQDVYGP  GSK3B   

                                           Kinase_seq  Length  answer  \
0   MKHKLGGGQYGEVYEGVWKKYSLTVAVKTLKEDTMEVEEFLKEAAV...     247       1   
1   MKHKLGGGQYGEVYEGVWKKYSLTVAVKTLKEDTMEVEEFLKEAAV...     247       1   
2   RQYRVLGKGGFGEVCACQVRATGKMYACKKLEKKRIKKRKGEAMAL...     261       1   
28  MKHKLGGGQYGEVYEGVWKKYSLTVAVKTLKEDTMEVEEFLKEAAV...     247       1   
30  YTDTKVIGNGSFGVVYQAKLCDSGELVAIKKVLQDKRFKNRELQIM...     284       1   

                                               matrix  train  
0   [[[0.965  1.207  1.078  ... 0.9233 0.9233 1.20...      1  
1   [[[0.993  0.965  1.078  ... 0.886  1.078  1.03...      1  
2   [[[0.7417 1.137  1.097  ... 1.17   1.17   1.09...      1  
28  [[[1.229 1.123 0.836 ... 0.965 0.836 1.034]\n ...      1  
30  [[[1.193  0.865  0.981  ... 1.269  0.981  0.88...      1

103
445


Substrate_UniprotID Substrate_name    Substrate_seq Kinase  \
3              O14920    IKBKB, IKKB  AKELDQGSLCTSFVG   CHUK   
4              O14920    IKBKB, IKKB  DQGSLCTSFVGTLQY   CHUK   
5              O14920    IKBKB, IKKB  ELDQGSLCTSFVGTL  IKBKB   
6              O14920    IKBKB, IKKB  SKVRGPVSGSPDSMN  IKBKB   
7              O14920    IKBKB, IKKB  VRGPVSGSPDSMNAS  IKBKB   

                                          Kinase_seq  Length  answer  \
3  MRERLGTGGFGNVCLYQHRELDLKIAIKSCRLELSTKNRERWCHEI...     272       1   
4  MRERLGTGGFGNVCLYQHRELDLKIAIKSCRLELSTKNRERWCHEI...     272       1   
5  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       1   
6  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       1   
7  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       1   

                                              matrix  train  
3  [[[1.23   0.886  0.965  ... 1.165  1.229  1.03...      0  
4  [[[0.873  1.042  1.034  ... 1.123  1.042  1.19...      0  
5  [[[0.8345 0.984  0.673  ... 0.7866 0.8745 0.98...      0  
6  [[[0.822  1.199  0.9087 ... 0.822  0.965  0.78...      0  
7  [[[0.9087 1.17   0.7866 ... 0.7827 1.012  0.82...      0

342
445


In [45]:
testdata_non['matrix']

3      [[[1.23   0.886  0.965  ... 1.165  1.229  1.03...
4      [[[0.873  1.042  1.034  ... 1.123  1.042  1.19...
5      [[[0.8345 0.984  0.673  ... 0.7866 0.8745 0.98...
6      [[[0.822  1.199  0.9087 ... 0.822  0.965  0.78...
7      [[[0.9087 1.17   0.7866 ... 0.7827 1.012  0.82...
                             ...                        
885    [[[0.87   1.211  0.9473 ... 0.976  1.211  0.88...
886    [[[1.1875 1.269  0.981  ... 0.981  1.269  1.06...
887    [[[1.165  1.023  1.165  ... 1.165  1.023  1.21...
888    [[[1.037  1.165  1.037  ... 0.9087 1.165  0.95...
889    [[[0.88   1.079  1.115  ... 0.861  1.079  1.09...
Name: matrix, Length: 786, dtype: object

In [42]:
testdata_non = pd.concat([testdata_true_anwswer0, testdata_false_anwswer0], axis=0)
testdata_non.head(1)

# # # save data
with open('/home/hb/python/pp_prediction/test_Data_curation/testdata_non_duple_x.pickle','wb') as fw:
    pickle.dump(testdata_non, fw)

# # load data
with open('/home/hb/python/pp_prediction/test_Data_curation/testdata_non_duple_x.pickle', 'rb') as fr:
    testdata_non = pickle.load(fr)

In [28]:
os.getcwd()

'/home/hb/python/pp_prediction/EDA'

In [44]:
testdata_non.iloc[0,-2]

'[[[1.23   0.886  0.965  ... 1.165  1.229  1.034 ]\n  [1.137  0.7046 1.17   ... 0.942  1.035  0.9194]\n  [1.012  1.199  0.8345 ... 0.88   0.9087 0.7866]\n  ...\n  [0.     0.     0.     ... 0.     0.     0.    ]\n  [0.     0.     0.     ... 0.     0.     0.    ]\n  [0.     0.     0.     ... 0.     0.     0.    ]]]'

In [35]:
# answer = 1인 것(403개) 중에 preds value도 1인 것 찾기
testdata_true_preds1 = testdata_true[testdata_true['preds'] == 1]
display(testdata_true_preds1)
print(len(testdata_true_preds1))
print(len(testdata_true))

# answer = 1 & train = 1(중복) 인 것 중에 PREDS VAULE 1인 것 찾기
testdata_true_anwswer1_preds1 = testdata_true_anwswer1[testdata_true_anwswer1['preds'] == 1]
display(testdata_true_anwswer1_preds1)
print(len(testdata_true_anwswer1_preds1))
print(len(testdata_true_anwswer1))


# answer = 0 & train = 0(중복) 인 것 중에 PREDS VAULE 1인 것 찾기
testdata_true_anwswer0_preds1 = testdata_true_anwswer0[testdata_true_anwswer0['preds'] == 1]
display(testdata_true_anwswer0_preds1)
print(len(testdata_true_anwswer0_preds1))
print(len(testdata_true_anwswer0))


Substrate_UniprotID    Substrate_name    Substrate_seq   Kinase  \
1                O00401      N-WASP, WASL  RETSKVIYDFIEKTG     ABL1   
2                O14745  NHERF1, SLC9A3R1  PALVRSASSDTSEEL     GRK6   
3                O14920       IKBKB, IKKB  AKELDQGSLCTSFVG     CHUK   
4                O14920       IKBKB, IKKB  DQGSLCTSFVGTLQY     CHUK   
5                O14920       IKBKB, IKKB  ELDQGSLCTSFVGTL    IKBKB   
..                  ...               ...              ...      ...   
434              P08575             PTPRC  EMSKESESDSDESSD  CSNK2A2   
436              Q15256             PTPRR  EPFVSIPTPREKVAM    MAPK1   
441              Q9Y5B0             CTDP1  AGESLDQSMEEEEEE  CSNK2A2   
442              Q9Y5B0             CTDP1  TKAQRENSPAAFPDR  CSNK2A2   
443              O60825            PFKFB2  PVRMRRNSFTPLSSS   PRKAA2   

                                            Kinase_seq  Length  answer  \
1    MKHKLGGGQYGEVYEGVWKKYSLTVAVKTLKEDTMEVEEFLKEAAV...     247       1   
2    RQYRVLGKGGFGEVCACQVRATGKMYACKKLEKKRIKKRKGEAMAL...     261       1   
3    MRERLGTGGFGNVCLYQHRELDLKIAIKSCRLELSTKNRERWCHEI...     272       1   
4    MRERLGTGGFGNVCLYQHRELDLKIAIKSCRLELSTKNRERWCHEI...     272       1   
5    EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       1   
..                                                 ...     ...     ...   
434  YQLVRKLGRGKYSEVFEAINITNNERVVVKILKPVKKKKIKREVKI...     285       1   
436  YTNLSYIGEGAYGMVCSAYDNVNKVRVAIKKISPFEHQTYCQRTLR...     288       1   
441  YQLVRKLGRGKYSEVFEAINITNNERVVVKILKPVKKKKIKREVKI...     285       1   
442  YQLVRKLGRGKYSEVFEAINITNNERVVVKILKPVKKKKIKREVKI...     285       1   
443  VLGDTLGVGTFGKVKIGEHQLTGHKVAVKILNRQKIRSLDVVGKIK...     251       1   

                                                matrix  probs  preds  train  
1    [[[0.993  0.965  1.078  ... 0.886  1.078  1.03...    1.0      1      1  
2    [[[0.7417 1.137  1.097  ... 1.17   1.17   1.09...    1.0      1      1  
3    [[[1.23   0.886  0.965  ... 1.165  1.229  1.03...    1.0      1      0  
4    [[[0.873  1.042  1.034  ... 1.123  1.042  1.19...    1.0      1      0  
5    [[[0.8345 0.984  0.673  ... 0.7866 0.8745 0.98...    1.0      1      0  
..                                                 ...    ...    ...    ...  
434  [[[0.865  1.193  1.023  ... 1.023  1.023  0.84...    1.0      1      0  
436  [[[0.865  0.886  1.211  ... 1.165  1.124  1.19...    1.0      1      1  
441  [[[1.124  0.981  0.865  ... 0.865  0.865  0.86...    1.0      1      1  
442  [[[1.027  1.013  1.124  ... 0.886  0.8477 1.06...    1.0      1      1  
443  [[[0.8213 1.498  1.035  ... 0.979  0.979  0.97...    1.0      1      0  

[228 rows x 11 columns]

228
445


Substrate_UniprotID                         Substrate_name  \
1                O00401                           N-WASP, WASL   
2                O14745                       NHERF1, SLC9A3R1   
30               O60716                   CTNND1, p120 catenin   
31               O60716                   CTNND1, p120 catenin   
33               O60716                   CTNND1, p120 catenin   
60               P02686                                    MBP   
61               P02686                                    MBP   
86               P02686                                    MBP   
106              P07947                              YES, YES1   
107              P07948                                    LYN   
108              P07948                                    LYN   
114              P08100                         RHO, Rhodopsin   
115              P08100                         RHO, Rhodopsin   
120              P08670                          VIM, Vimentin   
121              P08670                          VIM, Vimentin   
134              P10636                              MAPT, TAU   
160              P16220                                  CREB1   
161              P16949                        STMN1, Stathmin   
187              P20936                             GAP, RASA1   
209              P27986  PI3K regulatory subunit alpha, PIK3R1   
213              P29320                            EPHA3, hEK4   
214              P29320                            EPHA3, hEK4   
229              P29475                             NOS1, nNOS   
253              P37840                        SNCA, alpha-Syn   
274              P41743                      PKC-lambda, PRKCI   
278              P41743                      PKC-lambda, PRKCI   
280              P42224                                  STAT1   
282              P42229                                 STAT5A   
285              P42768                              WAS, WASp   
307              P53396                              ACL, ACLY   
315              P53779                           JNK3, MAPK10   
318              P53805                  Calcipressin-1, RCAN1   
330              Q02790                                  FKBP4   
332              Q03135                       CAV1, Caveolin-1   
335              Q05193                        DNM1, Dynamin-1   
343              Q05655                       PKC-delta, PRKCD   
346              Q05655                       PKC-delta, PRKCD   
348              Q05655                       PKC-delta, PRKCD   
352              Q12798                       CETN1, Centrin-1   
353              Q13153                                   PAK1   
354              Q13153                                   PAK1   
359              Q13480                                   GAB1   
365              Q13480                                   GAB1   
366              Q13480                                   GAB1   
370              Q13480                                   GAB1   
373              Q13480                                   GAB1   
377              Q13554           CAMK2B, CaMK-II subunit beta   
381              Q15418                       RPS6KA1, p90RSK1   
405              Q9GZM8                           NDEL1, Nudel   
406              Q9GZM8                           NDEL1, Nudel   
408              Q9GZM8                           NDEL1, Nudel   
411              Q9NRY4           ARHGAP35, GRLF1, p190 RhoGAP   
413              Q9UD71                       DARPP32, PPP1R1B   
419              Q9UD71                       DARPP32, PPP1R1B   
422              Q9UPZ9                               ICK, MRK   
425              P28562                                  DUSP1   
436              Q15256                                  PTPRR   
441              Q9Y5B0                                  CTDP1   
442              Q9Y5B0                                  CTDP1   

       Substrate_seq    Kinase  \
1    RETS

59
103


Substrate_UniprotID Substrate_name    Substrate_seq   Kinase  \
3                O14920    IKBKB, IKKB  AKELDQGSLCTSFVG     CHUK   
4                O14920    IKBKB, IKKB  DQGSLCTSFVGTLQY     CHUK   
5                O14920    IKBKB, IKKB  ELDQGSLCTSFVGTL    IKBKB   
9                O14920    IKBKB, IKKB  SPDSMNASRLSQPGQ    IKBKB   
10               O14920    IKBKB, IKKB  SMNASRLSQPGQLMS    IKBKB   
..                  ...            ...              ...      ...   
428              P08575          PTPRC  ESEHDSDSSSDDDSD  CSNK2A1   
432              P08575          PTPRC  ESEHDSDSSSDDDSD  CSNK2A2   
433              P08575          PTPRC  DSDESSDSDSDSEEP  CSNK2A2   
434              P08575          PTPRC  EMSKESESDSDESSD  CSNK2A2   
443              O60825         PFKFB2  PVRMRRNSFTPLSSS   PRKAA2   

                                            Kinase_seq  Length  answer  \
3    MRERLGTGGFGNVCLYQHRELDLKIAIKSCRLELSTKNRERWCHEI...     272       1   
4    MRERLGTGGFGNVCLYQHRELDLKIAIKSCRLELSTKNRERWCHEI...     272       1   
5    EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       1   
9    EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       1   
10   EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       1   
..                                                 ...     ...     ...   
428  YQLVRKLGRGKYSEVFEAINITNNEKVVVKILKPVKKKKIKREIKI...     285       1   
432  YQLVRKLGRGKYSEVFEAINITNNERVVVKILKPVKKKKIKREVKI...     285       1   
433  YQLVRKLGRGKYSEVFEAINITNNERVVVKILKPVKKKKIKREVKI...     285       1   
434  YQLVRKLGRGKYSEVFEAINITNNERVVVKILKPVKKKKIKREVKI...     285       1   
443  VLGDTLGVGTFGKVKIGEHQLTGHKVAVKILNRQKIRSLDVVGKIK...     251       1   

                                                matrix  probs  preds  train  
3    [[[1.23   0.886  0.965  ... 1.165  1.229  1.03...    1.0      1      0  
4    [[[0.873  1.042  1.034  ... 1.123  1.042  1.19...    1.0      1      0  
5    [[[0.8345 0.984  0.673  ... 0.7866 0.8745 0.98...    1.0      1      0  
9    [[[0.822  0.652  0.673  ... 0.652  0.7866 0.89...    1.0      1      0  
10   [[[0.822  0.965  0.7827 ... 0.984  0.965  0.82...    1.0      1      0  
..                                                 ...    ...    ...    ...  
428  [[[0.865  1.023  0.865  ... 0.8477 1.023  0.84...    1.0      1      0  
432  [[[0.865  1.023  0.865  ... 0.8477 1.023  0.84...    1.0      1      0  
433  [[[0.8477 1.023  0.8477 ... 0.865  0.865  0.88...    1.0      1      0  
434  [[[0.865  1.193  1.023  ... 1.023  1.023  0.84...    1.0      1      0  
443  [[[0.8213 1.498  1.035  ... 0.979  0.979  0.97...    1.0      1      0  

[169 rows x 11 columns]

169
342


#### 전체 445개의 테스트 데이터 중 prediction value를 1로 맞추는 비율이 어느정도인지

228개(pred = 1) / 445개(ALL) = 0.51

### 훈련데이터와 중복되는 103개의 테스트 데이터 중 prediction value를 1로 맞추는 비율이 어느정도인지

59개(preds = 1) / 103개(중복O) = 0.57

### 342개(445개-중복103개)의 테스트 데이터 중 prediction value를 1로 맞추는 비율이 어느정도인지

169개(preds = 1) / 342개(중복X) = 0.49

In [23]:
# 트레인과 중복되는 테스트 데이터
testdata_false_anwswer1 = testdata_false[testdata_false['train'] == 1]
display(testdata_false_anwswer1.head(5))
print(len(testdata_false_anwswer1))
print(len(testdata_false))

# 트레인과 중복되지 않는 테스트 데이터
testdata_false_anwswer0 = testdata_false[testdata_false['train'] == 0]
display(testdata_false_anwswer0.head(5))
print(len(testdata_false_anwswer0))
print(len(testdata_false))

Substrate_UniprotID Substrate_name    Substrate_seq Kinase  \
520              P02686            MBP  RTTTPTRTPTATATS  PRKCA   

                                            Kinase_seq  Length  answer  \
520  FLMVLGKGSFGKVMLADRKGTEELYAIKILKKDVVIQDDDVECTMV...     242       0   

                                                matrix  train  
520  [[[0.942 1.033 1.033 ... 1.115 1.033 1.079]\n ...      1

1
445


Substrate_UniprotID    Substrate_name    Substrate_seq Kinase  \
445              O00401      N-WASP, WASL  EYGYGYPYYYIYGYQ   ABL1   
446              O00401      N-WASP, WASL  PSDSDSTSYSPSKSG   ABL1   
447              O14745  NHERF1, SLC9A3R1  FYKYVYEYPYPYRYG   GRK6   
448              O14920       IKBKB, IKKB  GYPYSYSYSYIYDYL   CHUK   
449              O14920       IKBKB, IKKB  RSPSPSGSPSKSRSA   CHUK   

                                            Kinase_seq  Length  answer  \
445  MKHKLGGGQYGEVYEGVWKKYSLTVAVKTLKEDTMEVEEFLKEAAV...     247       0   
446  MKHKLGGGQYGEVYEGVWKKYSLTVAVKTLKEDTMEVEEFLKEAAV...     247       0   
447  RQYRVLGKGGFGEVCACQVRATGKMYACKKLEKKRIKKRKGEAMAL...     261       0   
448  MRERLGTGGFGNVCLYQHRELDLKIAIKSCRLELSTKNRERWCHEI...     272       0   
449  MRERLGTGGFGNVCLYQHRELDLKIAIKSCRLELSTKNRERWCHEI...     272       0   

                                                matrix  train  
445  [[[0.965 1.193 1.034 ... 1.034 1.193 1.042]\n ...      0  
446  [[[0.836  1.     0.873  ... 0.886  1.     1.03...      0  
447  [[[0.942  1.062  0.7046 ... 0.98   1.062  0.91...      0  
448  [[[1.034  1.193  0.836  ... 0.873  1.193  1.12...      0  
449  [[[0.993  1.     0.836  ... 0.993  1.     1.23...      0

444
445


In [73]:
# answer = 0 & train = 0(중복) 인 것 중에 PREDS VAULE 0인 것 찾기
testdata_false_preds0 = testdata_false[testdata_false['preds'] == 0]
display(testdata_false_preds0)
print(len(testdata_false_preds0))
print(len(testdata_false))


Substrate_UniprotID Substrate_name    Substrate_seq   Kinase  \
445              O00401   N-WASP, WASL  EYGYGYPYYYIYGYQ     ABL1   
446              O00401   N-WASP, WASL  PSDSDSTSYSPSKSG     ABL1   
448              O14920    IKBKB, IKKB  GYPYSYSYSYIYDYL     CHUK   
450              O14920    IKBKB, IKKB  SYLYTYPYNYRYQYP    IKBKB   
451              O14920    IKBKB, IKKB  SSVSLSGSLSASDSP    IKBKB   
..                  ...            ...              ...      ...   
882              Q15256          PTPRR  TSRSESESVSPSRSL    MAPK3   
883              Q15256          PTPRR  ESKSLSESDSLSQSE    MAPK3   
885              Q15256          PTPRR  PYQYIYFYRYEYKYE   PRKACA   
887              Q9Y5B0          CTDP1  VSVSGSLSFSFSVSF  CSNK2A2   
888              O60825         PFKFB2  GYGYVYRYGYVYEYQ   PRKAA2   

                                            Kinase_seq  Length  answer  \
445  MKHKLGGGQYGEVYEGVWKKYSLTVAVKTLKEDTMEVEEFLKEAAV...     247       0   
446  MKHKLGGGQYGEVYEGVWKKYSLTVAVKTLKEDTMEVEEFLKEAAV...     247       0   
448  MRERLGTGGFGNVCLYQHRELDLKIAIKSCRLELSTKNRERWCHEI...     272       0   
450  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       0   
451  EMKERLGTGGFGNVIRWHNQETGEQIAIKQCRQELSPRNRERWCLE...     274       0   
..                                                 ...     ...     ...   
882  YTQLQYIGEGAYGMVSSAYDHVRKTRVAIKKISPFEHQTYCQRTLR...     288       0   
883  YTQLQYIGEGAYGMVSSAYDHVRKTRVAIKKISPFEHQTYCQRTLR...     288       0   
885  FERIKTLGTGSFGRVMLVKHKETGNHYAMKILDKQKVVKLKQIEHT...     254       0   
887  YQLVRKLGRGKYSEVFEAINITNNERVVVKILKPVKKKKIKREVKI...     285       0   
888  VLGDTLGVGTFGKVKIGEHQLTGHKVAVKILNRQKIRSLDVVGKIK...     251       0   

                                                matrix         probs  preds  \
445  [[[0.965 1.193 1.034 ... 1.034 1.193 1.042]\n ...  5.913400e-09      0   
446  [[[0.836  1.     0.873  ... 0.886  1.     1.03...  1.980793e-07      0   
448  [[[1.034  1.193  0.836  ... 0.873  1.193  1.12...  5.423444e-06      0   
450  [[[0.822  0.865  0.984  ... 0.8994 0.865  0.65...  2.310642e-13      0   
451  [[[0.822  0.822  0.9087 ... 0.673  0.822  0.65...  7.736446e-11      0   
..                                                 ...           ...    ...   
882  [[[1.027  1.023  1.062  ... 1.062  1.023  1.15...  2.499768e-06      0   
883  [[[0.865  1.023  1.013  ... 1.003  1.023  0.86...  3.813407e-09      0   
885  [[[0.87   1.211  0.9473 ... 0.976  1.211  0.88...  6.501069e-09      0   
887  [[[1.165  1.023  1.165  ... 1.165  1.023  1.21...  4.642127e-09      0   
888  [[[1.037  1.165  1.037  ... 0.9087 1.165  0.95...  1.776588e-07      0   

     train  
445      0  
446      0  
448      0  
450      0  
451      0  
..     ...  
882      0  
883      0  
885      0  
887      0  
888      0  

[228 rows x 11 columns]

228
445


In [37]:
# answer

testdata_false_answer1 = testdata_false[testdata_false['train'] == 1]
display(testdata_false1.head(5))
print(len(testdata_false1))
print(len(testdata_false))

Substrate_UniprotID Substrate_name    Substrate_seq Kinase  \
520              P02686            MBP  RTTTPTRTPTATATS  PRKCA   

                                            Kinase_seq  Length  answer  \
520  FLMVLGKGSFGKVMLADRKGTEELYAIKILKKDVVIQDDDVECTMV...     242       0   

                                                matrix  train  
520  [[[0.942 1.033 1.033 ... 1.115 1.033 1.079]\n ...      1

1
445


In [38]:
testdata_true.to_csv("testdata_true.csv")
testdata_false.to_csv("testdata_false.csv")


In [39]:
testdata_true1

testdata_true1.loc[:,'subseq_kinseq'] = 0
testdata_true1['subseq_kinseq'] = testdata_true1['Substrate_seq'].str.cat(testdata_true1['Kinase_seq'], sep='_')

/home/hb/anaconda3/lib/python3.9/site-packages/pandas/core/indexing.py:1667: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.obj[key] = value
/tmp/ipykernel_1308621/701656937.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  testdata_true1['subseq_kinseq'] = testdata_true1['Substrate_seq'].str.cat(testdata_true1['Kinase_seq'], sep='_')


In [40]:
testdata_true1.head(3)

Substrate_UniprotID    Substrate_name    Substrate_seq Kinase  \
0              O00401      N-WASP, WASL  EITTNRFYGPQVNNI   ABL1   
1              O00401      N-WASP, WASL  RETSKVIYDFIEKTG   ABL1   
2              O14745  NHERF1, SLC9A3R1  PALVRSASSDTSEEL   GRK6   

                                          Kinase_seq  Length  answer  \
0  MKHKLGGGQYGEVYEGVWKKYSLTVAVKTLKEDTMEVEEFLKEAAV...     247       1   
1  MKHKLGGGQYGEVYEGVWKKYSLTVAVKTLKEDTMEVEEFLKEAAV...     247       1   
2  RQYRVLGKGGFGEVCACQVRATGKMYACKKLEKKRIKKRKGEAMAL...     261       1   

                                              matrix  train  \
0  [[[0.965  1.207  1.078  ... 0.9233 0.9233 1.20...      1   
1  [[[0.993  0.965  1.078  ... 0.886  1.078  1.03...      1   
2  [[[0.7417 1.137  1.097  ... 1.17   1.17   1.09...      1   

                                       subseq_kinseq  
0  EITTNRFYGPQVNNI_MKHKLGGGQYGEVYEGVWKKYSLTVAVKTL...  
1  RETSKVIYDFIEKTG_MKHKLGGGQYGEVYEGVWKKYSLTVAVKTL...  
2  PALVRSASSDTSEEL_RQYRVLGKGGFGEVCACQVRATGKMYACKK...

In [42]:
subseq_kinseq = testdata_true1.iloc[:,-1]
subseq_kinseq

0      EITTNRFYGPQVNNI_MKHKLGGGQYGEVYEGVWKKYSLTVAVKTL...
1      RETSKVIYDFIEKTG_MKHKLGGGQYGEVYEGVWKKYSLTVAVKTL...
2      PALVRSASSDTSEEL_RQYRVLGKGGFGEVCACQVRATGKMYACKK...
28     VLPQDKEYYKVKEPG_MKHKLGGGQYGEVYEGVWKKYSLTVAVKTL...
30     MEGYRAPSRQDVYGP_YTDTKVIGNGSFGVVYQAKLCDSGELVAIK...
                             ...                        
438    EPFVSIPTPREKVAM_YTQLQYIGEGAYGMVSSAYDHVRKTRVAIK...
439    GLQERRGSNVSLTLD_FERIKTLGTGSFGRVMLVKHKETGNHYAMK...
441    AGESLDQSMEEEEEE_YQLVRKLGRGKYSEVFEAINITNNERVVVK...
442    TKAQRENSPAAFPDR_YQLVRKLGRGKYSEVFEAINITNNERVVVK...
444    TPLSSSNTIRRPRNY_FLMVLGKGSFGKVMLADRKGTEELYAIKIL...
Name: subseq_kinseq, Length: 103, dtype: object

In [74]:
traindata

Substrate_UniprotID Substrate_name    Substrate_seq Kinase  \
0                  Q9ERR1          NDEL1  LGSSRPSSAPGMLPL  DYRK2   
1                  Q01986           MEK1  RTPGRPLSSYGMDSR   PAK2   
2                  Q9WTQ1          PRKD1  GVRRRRLSNVSLTGL   PAK2   
3                  P01237      prolactin  IRCLRRDSHKVDNYL   PAK2   
4                  P01236      prolactin  LHCLRRDSHKIDNYL   PAK2   
...                   ...            ...              ...    ...   
35437              Q8N122         Raptor  STKTGTSTKTKTATF   ULK2   
35438              Q8N122         Raptor  AYIYRYMYSYPYSYD   ULK2   
35439              O15027         SEC16A  GYSYHYGYKYTYLYK   ULK2   
35440              Q8NEB9         PIK3C3  QSTSQSASDSMSKST   ULK2   
35441              A2RUS2         DENND3  VYKYIYEYTYNYEYH   ULK2   

                                              Kinase_seq  Length  answer  \
0      YEVLKVIGKGSFGQVVKAYDHKVHQHVALKMVRNEKRFHRQAAEEI...     313       1   
1      YTRYEKIGQGASGTVFTATDVALGQEVAIKQINLQKQPKKELIINE...     251       1   
2      YTRYEKIGQGASGTVFTATDVALGQEVAIKQINLQKQPKKELIINE...     251       1   
3      YTRYEKIGQGASGTVFTATDVALGQEVAIKQINLQKQPKKELIINE...     251       1   
4      YTRYEKIGQGASGTVFTATDVALGQEVAIKQINLQKQPKKELIINE...     251       1   
...                                                  ...     ...     ...   
35437  YSKRDLVGHGAFAVVFRGRHRQKTDWEVAIKSINKKNLSKSQILLG...     262       0   
35438  YSKRDLVGHGAFAVVFRGRHRQKTDWEVAIKSINKKNLSKSQILLG...     262       0   
35439  YSKRDLVGHGAFAVVFRGRHRQKTDWEVAIKSINKKNLSKSQILLG...     262       0   
35440  YSKRDLVGHGAFAVVFRGRHRQKTDWEVAIKSINKKNLSKSQILLG...     262       0   
35441  YSKRDLVGHGAFAVVFRGRHRQKTDWEVAIKSINKKNLSKSQILLG...     262       0   

                                                  matrix  
0      [[[1.152  0.981  1.023  ... 1.152  0.886  1.15...  
1      [[[1.062  1.027  0.886  ... 0.8477 1.023  1.06...  
2      [[[0.981  1.165  1.062  ... 1.027  0.981  1.15...  
3      [[[1.1875 1.062  1.37   ... 1.003  1.269  1.15...  
4      [[[1.152  1.063  1.37   ... 1.003  1.269  1.15...  
...                                                  ...  
35437  [[[1.023  1.027  1.013  ... 1.124  1.027  1.21...  
35438  [[[1.124  1.269  1.1875 ... 1.023  1.269  0.84...  
35439  [[[0.981  1.269  1.023  ... 1.152  1.269  1.01...  
35440  [[[1.003  1.023  1.027  ... 1.013  1.023  1.02...  
35441  [[[1.165  1.269  1.013  ... 0.865  1.269  1.06...  

[35442 rows x 8 columns]

In [112]:
'YTQLQYIGEGAYGMVSSAYDHVRKTRVAIKKISPFEHQTYCQRTLREIQILLRFRHENVIGIRDILRASTLEAMRDVYIVQDLMETDLYKLLKSQQLSNDHICYFLYQILRGLKYIHSANVLHRDLKPSNLLINTTCDLKICDFGLARIADPEHDHTGFLTEYVATRWYRAPEIMLNSKGYTKSIDIWSVGCILAEMLSNRPIFPGKHYLDQLNHILGILGSPSQEDLNCIINMKARNYLQSLPSKTKVAWAKLFPKSDSKALDLLDRMLTFNPNKRITVEEALAHPY000000000000000000000000000000000000000000000000' == 'YTQLQYIGEGAYGMVSSAYDHVRKTRVAIKKISPFEHQTYCQRTLREIQILLRFRHENVIGIRDILRAPTLEAMRDVYIVQDLMETDLYKLLKSQQLSNDHICYFLYQILRGLKYIHSANVLHRDLKPSNLLINTTCDLKICDFGLARIADPEHDHTGFLTEYVATRWYRAPEIMLNSKGYTKSIDIWSVGCILAEMLSNRPIFPGKHYLDQLNHILGILGSPSQEDLNCIINMKARNYLQSLPSKTKVAWAKLFPKSDSKALDLLDRMLTFNPNKRITVEEALAHPY000000000000000000000000000000000000000000000000'

False

In [107]:
print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'ERK1'].unique()))
print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'ERK2'].unique()))

print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'ERK3'].unique()))

print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'ERK5'].unique()))

print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'ERK7'].unique()))


['YTQLQYIGEGAYGMVSSAYDHVRKTRVAIKKISPFEHQTYCQRTLREIQILLRFRHENVIGIRDILRASTLEAMRDVYIVQDLMETDLYKLLKSQQLSNDHICYFLYQILRGLKYIHSANVLHRDLKPSNLLINTTCDLKICDFGLARIADPEHDHTGFLTEYVATRWYRAPEIMLNSKGYTKSIDIWSVGCILAEMLSNRPIFPGKHYLDQLNHILGILGSPSQEDLNCIINMKARNYLQSLPSKTKVAWAKLFPKSDSKALDLLDRMLTFNPNKRITVEEALAHPY000000000000000000000000000000000000000000000000', 'YTQLQYIGEGAYGMVSSAYDHVRKTRVAIKKISPFEHQTYCQRTLREIQILLRFRHENVIGIRDILRAPTLEAMRDVYIVQDLMETDLYKLLKSQQLSNDHICYFLYQILRGLKYIHSANVLHRDLKPSNLLINTTCDLKICDFGLARIADPEHDHTGFLTEYVATRWYRAPEIMLNSKGYTKSIDIWSVGCILAEMLSNRPIFPGKHYLDQLNHILGILGSPSQEDLNCIINMKARNYLQSLPSKTKVAWAKLFPKSDSKALDLLDRMLTFNPNKRITVEEALAHPY000000000000000000000000000000000000000000000000']
['YTNLSYIGEGAYGMVCSAYDNLNKVRVAIKKISPFEHQTYCQRTLREIKILLRFRHENIIGINDIIRAPTIEQMKDVYIVQDLMETDLYKLLKTQHLSNDHICYFLYQILRGLKYIHSANVLHRDLKPSNLLLNTTCDLKICDFGLARVADPDHDHTGFLTEYVATRWYRAPEIMLNSKGYTKSIDIWSVGCILAEMLSNRPIFPGKHYLDQLNHILGILGSPSQEDLNCIINLKARNYLLSLPHKNKVPWNRLFPNADSKALDLLDKMLTFNPHKRIEVEQALAHPY00000000000000000000000000000

In [108]:
print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'CLK1'].unique()))

print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'CLK2'].unique()))

print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'CLK3'].unique()))

print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'CLK4'].unique()))

['YEIVDTLGEGAFGKVVECIDHKAGGRHVAVKIVKNVDRYCEAARSEIQVLEHLNTTDPNSTFRCVQMLEWFEHHGHICIVFELLGLSTYDFIKENGFLPFRLDHIRKMAYQICKSVNFLHSNKLTHTDLKPENILFVQSDYTEAYNPKIKRDERTLINPDIKVVDFGSATYDDEHHSTLVSTRHYRAPEVILALGWSQPCDVWSIGCILIEYYLGFTVFPTHDSKEHLAMMERILGPLPKHMIQKTRKRKYFHHDRLDWDEHSSAGRYVSRRCKPLKEFMLSQDVEHERLFDLIQKMLEYDPAKRITLREALKHPF00000000000000000000']
['YEIVSTLGEGTFGRVVQCVDHRRGGARVALKIIKNVEKYKEAARLEINVLEKINEKDPDNKNLCVQMFDWFDYHGHMCISFELLGLSTFDFLKDNNYLPYPIHQVRHMAFQLCQAVKFLHDNKLTHTDLKPENILFVNSDYELTYNLEKKRDERSVKSTAVRVVDFGSATFDHEHHSTIVSTRHYRAPEVILELGWSQPCDVWSIGCIIFEYYVGFTLFQTHDNREHLAMMERILGPIPSRMIRKTRKQKYFYRGRLDWDENTSAGRYVRENCKPLRRYLTSEAEEHHQLFDLIESMLEYEPAKRLTLGEALQHPF00000000000000000000']
['YEIVGNLGEGTFGKVVECLDHARGKSQVALKIIRNVGKYREAARLEINVLKKIKEKDKENKFLCVLMSDWFNFHGHMCIAFELLGKNTFEFLKENNFQPYPLPHVRHMAYQLCHALRFLHENQLTHTDLKPENILFVNSEFETLYNEHKSCEEKSVKNTSIRVADFGSATFDHEHHTTIVATRHYRPPEVILELGWAQPCDVWSIGCILFEYYRGFTLFQTHENREHLVMMEKILGPIPSHMIHRTRKQKYFYKGGLVWDENSSDGRYVKENCKPLKSYMLQDSLEHVQLFDLMRRMLEFDPAQRITLAEALLHPF

In [109]:
print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'GRK1'].unique()))

print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'GRK3'].unique()))

print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'GRK4'].unique()))

print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'GRK5'].unique()))
print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'GRK6'].unique()))
print(list(traindata['Kinase_seq'][traindata['Kinase'] == 'GRK7'].unique()))

['FRVLGRGGFGEVFACQMKATGKLYACKKLNKKRLKKRKGYQGAMVEKKILAKVHSRFIVSLAYAFETKTDLCLVMTIMNGGDIRYHIYNVDEDNPGFQEPRAIFYTAQIVSGLEHLHQRNIIYRDLKPENVLLDDDGNVRISDLGLAVELKAGQTKTKGYAGTPGFMAPELLLGEEYDFSVDYFALGVTLYEMIAARGPFRARGEKVENKELKQRVLEQAVTYPDKFSPASKDFCEALLQKDPEKRLGFR00000000000000000000000000000000000000000000000000000000000000000000000000000000000000', 'FRVLGKGGFGEVSACQMKATGKLYACKKLNKKRLKKRKGYQGAMVEKKILMKVHSRFIVSLAYAFETKADLCLVMTIMNGGDIRYHIYNVNEENPGFPEPRALFYTAQIICGLEHLHQRRIVYRDLKPENVLLDNDGNVRISDLGLAVELLDGQSKTKGYAGTPGFMAPELLQGEEYDFSVDYFALGVTLYEMIAARGPFRARGEKVENKELKHRIISEPVKYPDKFSQASKDFCEALLEKDPEKRLGFR00000000000000000000000000000000000000000000000000000000000000000000000000000000000000', 'FRVLGKGGFGEVSACQMKATGKMYACKKLNKKRLKKKKGYQGAIVEKRILTKVHSRFIVSLAYAFETKTDLCLVMTIMNGGDVRYHIYNVDEDNPGFSEPRAIYYTAQIISGLEHLHQRRIVYRDLKPENVLLDNDGNIRISDLGLAVELKEGQNKTKGYAGTPGFMAPELLRGEEYDFSVDYFALGVTLYEMIAARGPFRARGEKVENKELKQRIISEPVKYPEKFSQASKDFCEQLLEKDPEKRLGFR00000000000000000000000000000000000000000000000000000000000000000000

In [110]:
'FRVLGRGGFGEVFACQMKATGKLYACKKLNKKRLKKRKGYQGAMVEKKILAKVHSRFIVSLAYAFETKTDLCLVMTIMNGGDIRYHIYNVDEDNPGFQEPRAIFYTAQIVSGLEHLHQRNIIYRDLKPENVLLDDDGNVRISDLGLAVELKAGQTKTKGYAGTPGFMAPELLLGEEYDFSVDYFALGVTLYEMIAARGPFRARGEKVENKELKQRVLEQAVTYPDKFSPASKDFCEALLQKDPEKRLGFR00000000000000000000000000000000000000000000000000000000000000000000000000000000000000' == 'FRVLGKGGFGEVSACQMKATGKLYACKKLNKKRLKKRKGYQGAMVEKKILMKVHSRFIVSLAYAFETKADLCLVMTIMNGGDIRYHIYNVNEENPGFPEPRALFYTAQIICGLEHLHQRRIVYRDLKPENVLLDNDGNVRISDLGLAVELLDGQSKTKGYAGTPGFMAPELLQGEEYDFSVDYFALGVTLYEMIAARGPFRARGEKVENKELKHRIISEPVKYPDKFSQASKDFCEALLEKDPEKRLGFR00000000000000000000000000000000000000000000000000000000000000000000000000000000000000'

False

In [86]:
traindata['Kinase'].unique()

array(['DYRK2', 'PAK2', 'PKCD', 'Pim2', 'CK2A1', 'AurA', 'IKKB', 'DRAK2',
       'ITK', 'VRK1', 'CDK7', 'FGFR4', 'RIPK1', 'JNK1', 'DYRK4',
       'FAK iso2', 'EGFR', 'Fyn', 'AMPKA1', 'CDK12', 'CAMK4', 'AMPKA2',
       'NDR2', 'CK1E', 'MAPKAPK2', 'CAMK1A', 'Yes', 'Akt1', 'PKCB iso2',
       'EphB2', 'TBK1', 'HIPK3', 'TIE2', 'PRKD3', 'CDK8', 'Syk', 'STLK3',
       'HUNK', 'LYN', 'MYO3A', 'PKCI', 'IRAK1', 'CDKL5', 'PKACB', 'P38B',
       'HER4', 'STYK1', 'CDK2', 'Ret', 'RAF1', 'caMLCK', 'GSK3B', 'GRK6',
       'MEKK1', 'MARK3 iso3', 'CK1G2', 'Akt2', 'PKCT', 'FGR', 'CK1G1',
       'LKB1', 'MEK5', 'INSR', 'EphB1', 'LTK', 'Chk1', 'PKACA', 'GRK3',
       'ERK5', 'MARK1', 'ERK1', 'DAPK3', 'RIPK3', 'HPK1', 'AurC',
       'CAMKK1', 'Pim3', 'MEK1', 'Fes', 'CK1D', 'MEKK3', 'KSR2', 'CAMK2B',
       'PLK4', 'PKCB', 'BRSK1', 'MKK7', 'FRK', 'MPSK1', 'PAK1', 'MEKK2',
       'SGK1', 'HER2', 'MRCKA', 'OSR1', 'Src', 'PLK3', 'ICK', 'Arg',
       'GRK5', 'CDK3', 'WNK2', 'TrkB', 'ILK', 'TESK1', 'TIE1', 'PKCG